newIDE/app/src/EventsSheet/InstructionEditor/EventTextDialog.js

```diff
@@ -54,180 +48,171 @@ export const filterEditableWithEventTextDialog = (
-      //Background color
-      gd.asGroupEvent(event).setBackgroundColor(
-        backgroundColor.r,
-        backgroundColor.g,
-        backgroundColor.b
-      );
+const EventTextDialog = (props: Props) => {
+  const { event, onClose } = props;
+
+  const [textValue, setTextValue] = React.useState<string>('');
+  const [textColor, setTextColor] = React.useState<RGBColor>(black);
+  const [backgroundColor, setBackgroundColor] = React.useState<RGBColor>(black);
+
+  const eventType = event.getType(); // OK
+
+  React.useEffect(
+    () => {
+      if (eventType === 'BuiltinCommonInstructions::Comment') {
+        const commentEvent = gd.asCommentEvent(event);
+
+        setTextColor({
+          r: commentEvent.getTextColorRed(),
+          g: commentEvent.getTextColorGreen(),
+          b: commentEvent.getTextColorBlue(),
+        });
+
+        setBackgroundColor({
+          r: commentEvent.getBackgroundColorRed(),
+          g: commentEvent.getBackgroundColorGreen(),
+          b: commentEvent.getBackgroundColorBlue(),
+        });
+
+        setTextValue(gd.asCommentEvent(event).getComment());
+      } else if (eventType === 'BuiltinCommonInstructions::Group') {
+        var groupEvent = gd.asGroupEvent(event);
+        const r = groupEvent.getBackgroundColorR(),
+          g = groupEvent.getBackgroundColorG(),
+          b = groupEvent.getBackgroundColorB();
+
+        // Text color is automatically chosen for groups.
+        setTextColor(() => {
+          return (r + g + b) / 3 > 200 ? black : white;
+        });
+
+        setBackgroundColor({
+          r: groupEvent.getBackgroundColorR(),
+          g: groupEvent.getBackgroundColorG(),
+          b: groupEvent.getBackgroundColorB(),
+        });
+
+        setTextValue(gd.asGroupEvent(event).getName());
+      } else {
+        console.error(
+          'Dialog was opened for an unsupported event type: ' + eventType
+        );
+      }
+    },
+    [event, eventType]
+  );
+
+  const onApply = React.useCallback(
+    () => {
+      if (eventType === 'BuiltinCommonInstructions::Comment') {
+        //Text value
+        gd.asCommentEvent(event).setComment(textValue);
+
+        //Text color
+        gd.asCommentEvent(event).setTextColor(
+          textColor.r,
+          textColor.g,
+          textColor.b
+        );
+        //Background color
+        gd.asCommentEvent(event).setBackgroundColor(
+          backgroundColor.r,
+          backgroundColor.g,
+          backgroundColor.b
+        );
+      } else if (eventType === 'BuiltinCommonInstructions::Group') {
+        //Text value
+        gd.asGroupEvent(event).setName(textValue);
+
+        //Text color for group not supported in Core, instead GroupEvent.js handle this
+        //Background color
+        gd.asGroupEvent(event).setBackgroundColor(
+          backgroundColor.r,
+          backgroundColor.g,
+          backgroundColor.b
+        );
+      }
+      props.onApply();
```

Comment:
`onApply` is used with `props` because if not flow show an error, because `onApply` is already defined at line 105.

Reply:
Yes that's fine, these are two different onApply.